src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -178,4 +187,66 @@ protected CustomerInvoiceDetail createSourceAccountingLinesByContractControlAcco
                 coaCode, accountNumber, objectCode, getTotalAmountForInvoice(contractsGrantsInvoiceDocument), 1);
     }
 
+    /**
+     * Overridden to also update the invoice due date.
+     * 
+     * NOTE: There isn't an easy way to hook in the invoice-due-date updates independently of this
+     * without overlaying the CINV document class. (This method only appears to be called when CINV docs
+     * enter PROCESSED status, so hooking in here is okay for now.) If ContractsGrantsInvoiceDocument
+     * gets updated to make it easier to hook in custom PROCESSED-status handling, then we should rework
+     * this customization.
+     */
+    @Override
+    public void updateLastBilledDate(ContractsGrantsInvoiceDocument document) {
+        super.updateLastBilledDate(document);
+        setInvoiceDueDateBasedOnNetTermsAndCurrentDate(document);
+    }
+
+    @Override
+    public void setInvoiceDueDateBasedOnNetTermsAndCurrentDate(ContractsGrantsInvoiceDocument document) {
+        Date invoiceDueDate = calculateInvoiceDueDate(document);
+        document.setInvoiceDueDate(invoiceDueDate);
+    }
+
+    protected Date calculateInvoiceDueDate(ContractsGrantsInvoiceDocument document) {
+        Calendar calendar = dateTimeService.getCurrentCalendar();
+        Optional<Integer> customerNetTerms = getCustomerNetTerms(document);
+        int netTermsInDays;
+        
+        if (customerNetTerms.isPresent()) {
+            netTermsInDays = customerNetTerms.get();
+        } else {
+            String defaultNetTerms = parameterService.getParameterValueAsString(
+                    ArConstants.AR_NAMESPACE_CODE, ArConstants.CUSTOMER_COMPONENT, CuArParameterKeyConstants.CG_INVOICE_TERMS_DUE_DATE);
+            try {
+                netTermsInDays = Integer.parseInt(defaultNetTerms);
+            } catch (NumberFormatException e) {
+                LOG.error("calculateInvoiceDueDate, Default net terms parameter is malformed; will use zero as the default instead", e);
+                netTermsInDays = 0;
```

Comment:
It might be a good idea to check with the functionals what should happen if the parameter is improperly formatted

Reply:
Good point, I'll check to see what the functionals want the behavior to be here.

Reply:
The functionals have confirmed that an error should be thrown in this spot if the net terms parameter value is invalid. I'll update the code accordingly. Thanks for pointing this out!